src/bufferedBot.ts

```diff
@@ -37,14 +62,55 @@ export class BufferedBot extends TelegramBot {
       concatMap(compoundMessage => timer(SPACE_TIME).pipe(
         ignoreElements(),
         startWith(compoundMessage),
-      )),
-    ).subscribe(compoundMessage => {
-      log.debug('Sending compound message %o', compoundMessage)
+        mergeMap(compoundMessage => {
+          log.debug('Sending compound message %o', compoundMessage)
 
-      const { chatId, text, options } = compoundMessage
+          const { chatId, text, options } = compoundMessage
 
-      super.sendMessage(chatId, text, options)
-    })
+          return super.sendMessage(chatId, text, { ...options }) // {...hack} because bot mutates options
+        }),
+        retryWhen(error$ =>
+          error$.pipe(
+            switchMap((error: Error, index) => {
+              // on MAX_RETRIES error -- skip this unfortunate message
+              log.error('Error sending message:', error.message)
+              if (index === MAX_RETRIES) {
+                log.error('Maximum retries of', MAX_RETRIES, 'reached')
+                log.error('Message %o will not be sent', compoundMessage)
+                // rethrow error
+                return throwError(error)
+              }
+
+              // pass the error on
+              return of(error)
+            }),
+            map(error => {
+              // if error is 'Too Many Requests: retry in N'
+              const errorMatch = error.message.match(RETRY_IN_REGEXP)
+              const delaySec = errorMatch ? parseInt(errorMatch[1], 10) : SPACE_TIME
+              // or in SPACE_TIME seconds
+
+              log.error('Retrying in', delaySec, 'seconds')
+
+              return delaySec
+            }),
+            // delay between retries
+            delayWhen(delaySec => timer(delaySec * 1000)),
+            tap(() => log.error('Retrying now'), e => console.log('ErrorError', !!e), () => console.log('COMPLETE complete')),
```

Comment:
Review/remove all the console.logs before merging

Reply:
You are somehow reviewing an old commit
It's already removed, only `tap(() => log.error('Retrying now'))` stays as it is actually useful